nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/PublishAMQP.java

```diff
@@ -62,6 +64,7 @@
         + "and Queue is not set up, the message will have no final destination and will return (i.e., the data will not make it to the queue). If "
         + "that happens you will see a log in both app-log and bulletin stating to that effect. Fixing the binding "
         + "(normally done by AMQP administrator) will resolve the issue.")
+@HighResourceUsageScenario(resource = SystemResource.MEMORY)
```

Comment:
We need to be able to articulate the memory usage.  Is it that every message published is fully loaded into memory in a byte[] therefore large messages will consume large amounts of heap?  Same for a lot of items below.  We need to be able to let the developer explain.  In some cases we have processors that operate on batches of things and people will worry it is the batch that is the problem.  But in reality it is that if any single event/record is large within a batch that single event will be in mem/etc...

Reply:
The developer can provide a description using the "scenario" argument on the annotation.  This first pass was to identify most of the processors that have the annotation.  As we look through the list of components, specific descriptions can be added to override the default scenario from the annotation itself.